tests/integration/deploy/test_managed_stack_deploy.py

```diff
@@ -0,0 +1,120 @@
+            stack_name = stack["name"]
+            if stack_name != SAM_CLI_STACK_NAME:
+                region = stack.get("region")
+                cfn_client = (
+                    self.cfn_client if not region else boto3.client("cloudformation", config=Config(region_name=region))
+                )
+                cfn_client.delete_stack(StackName=stack_name)
+        super().tearDown()
+
+    @parameterized.expand(["aws-serverless-function.yaml"])
+    def test_managed_stack_creation(self, template_file):
+        self._delete_managed_stack(self.cfn_client, self.s3_client)
+        self.assertFalse(self._does_stack_exist(self.cfn_client, SAM_CLI_STACK_NAME))
+
+        template_path = self.test_data_path.joinpath(template_file)
+
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        deploy_command_list = self.get_deploy_command_list(
+            template_file=template_path,
+            stack_name=stack_name,
+            capabilities="CAPABILITY_IAM",
+            force_upload=True,
+            notification_arns=self.sns_arn,
+            parameter_overrides="Parameter=Clarity",
+            kms_key_id=self.kms_key,
+            tags="integ=true clarity=yes foo_bar=baz",
+            resolve_s3=True,
+        )
+
+        deploy_process_execute = run_command(deploy_command_list)
+        self.assertEqual(deploy_process_execute.process.returncode, 0)
+        self.assertTrue(self._does_stack_exist(self.cfn_client, SAM_CLI_STACK_NAME))
+
+    def _method_to_stack_name(self, method_name):
+        """Method expects method name which can be a full path. Eg: test.integration.test_deploy_command.method_name"""
+        method_name = method_name.split(".")[-1]
+        return f"{method_name.replace('_', '-')}-{CFN_PYTHON_VERSION_SUFFIX}"
+
+    def _delete_managed_stack(self, cfn_client, s3_client, wait=True):
+        if not self._does_stack_exist(cfn_client, SAM_CLI_STACK_NAME):
+            return
+
+        stack = boto3.resource("cloudformation").Stack(SAM_CLI_STACK_NAME)
+        resources = stack.resource_summaries.all()
+        for resource in resources:
+            if resource.resource_type == "AWS::S3::Bucket":
+                s3_bucket_name = resource.physical_resource_id
+
+        if s3_bucket_name:
+            s3 = boto3.resource("s3")
+            bucket = s3.Bucket(s3_bucket_name)
+            bucket.object_versions.delete()
+            s3_client.delete_bucket(Bucket=s3_bucket_name)
+        cfn_client.delete_stack(StackName=SAM_CLI_STACK_NAME)
+
+        if wait:
+            waiter = cfn_client.get_waiter("stack_delete_complete")
+            waiter_config = {"Delay": 15, "MaxAttempts": 120}
+            waiter.wait(StackName=SAM_CLI_STACK_NAME, WaiterConfig=waiter_config)
+
+    def _does_stack_exist(self, cfn_client, stack_name):
+        try:
+            cfn_client.describe_stacks(StackName=stack_name)
```

Comment:
Stack needs to exist and in a good state, not a ROLLBACK_COMPLETE state, lets check that too.

Reply:
Lets check for the resources to exist as well apart from the state of CFN stack.